flink-runtime/src/main/java/org/apache/flink/runtime/io/network/NetworkEnvironment.java

```diff
@@ -202,28 +233,97 @@ public TaskKvStateRegistry createKvStateTaskRegistry(JobID jobId, JobVertexID jo
 		return kvStateRegistry.createTaskRegistry(jobId, jobVertexId);
 	}
 
+	public ResultPartition[] getResultPartitionWriters(String taskId) {
```

Comment:
will the ShuffleService also have the partition/gate getters then? it seems logical to let ShuffleService manage partitions/gates. Alternatively, task could still keep final arrays with partitions/gates. Then there will be less dependencies on ShuffleService interface. What do you think?

Reply:
That is a good idea for removing gutters from `ShuffleService` to make interface simple, then let task still maintain the arrays.